docs/02_database/01_fields-tables-views/02_tables/01_tables-basics.md

```diff
@@ -100,38 +105,113 @@ tables {
+Imagine a scenario where you have projects and users.
+
+- You have a PROJECTS table where each project has a unique PROJ_ID.
+- You have a USERS table where each project has a unique USER_ID.
+
+There is a many-to-many relationship between the two tables - projects can have more than one user, and users can belong to more than one project. To allocate users to projects, you need a third table: PROJECT_USERS.
+
+In the the PROJECT_USERS table, you only need two fields:
+
+- PROJ_ID
+- USER_ID
+
+Each record in this table states that PROJ_ID= x has USER_ID=y. But it is important that you do not duplicate any of these records - which could cause problems. So, in your PROJECT_USERS table, make PROJ_ID the primary index and add a unique index on PROJ_ID and USER_ID. The database will now prevent you from entering a record that has exactly the same PROJ_ID and USER_ID as an existing record.
+
+Here is our table definition for this:
 
 ```kotlin
-    indices {
-        unique {
-            SESSION_AUTH_TOKEN
-        }
-        nonUnique {
-            USER_NAME
-        }
-    }
+  table(name = "PROJECT_USERS", id = 11020) {
+      PROJ_ID
+      USER_ID
+      primaryKey {
+          PROJ_ID
+      }
+     indices {
+         unique {
+             PROJ_ID
+             USER_ID
+         }
+         nonUnique {
+             USER_ID
+         }
+     }
+  }
 ```
+You can clearly see the unique index on the two fields in the table above. As well as enforcing uniqueness, it also enables you to search for a record where USER_ID=y and USER_ID=y - so you can see if project x includes user y (and you can also make a partial search - see below).
+
+Note that the example also includes a non-unique index on USER_ID. This enables you to search for all records where USER_ID=y - so you can see exactly which projects the user belongs to.
+
+### Indices with multiple fields
+Any index you create - the primary key or other unique or non-unique indices - can have multiple fields.
+
+If you create an index with multiple fields, it is possible to make a search based on all the fields specified or to make a partial search based on some of the fields. However, there are strict limits to this.
+
+Consider an index that has three fields; A, B and C:
+
+- The search **must** always use the first field: A. It is also possible to search only on this field.
+- The search **can** use all the fields. The field values must be specified in the order that they are specified in the index.
+- The search **can** use the first and second fields only: A and B.
+- The search **cannot** use the first and third fields (A and C) only.
+- The search **cannot** use the second and third fields (B and C) only.
+
+Bear these restrictions in mind when you create an index with multiple fields; the order of the fields is important if you want partial searches to be useful and efficient.
```

Comment:
I think the wording in the indices file is clear, it might be better to refer to that definition, rather than having two ways of explaining the same thing.